### DISCUSSION TITLE
Error: Failed to launch the browser process!

### DISCUSSION THREAD

**Discussion by yannisc** [spatie/laravel-pdf#57] [Q&A] [answered]
I have a working installation on my Mac, but when trying to run the pdf creation process on AlmaLinux 8, I get this error:

The command "PATH=$PATH:/usr/local/bin:/opt/homebrew/bin NODE_PATH=`npm root -g` node '/home/erp/public_html/releases/20240123132720/vendor/spatie/browsershot/src/../bin/browser.cjs' '{"url":"file:\/\/\/tmp\/1446216859-0747839001706019855\/index.html","action":"pdf","options":{"args":[],"viewport":{"width":800,"height":600},"displayHeaderFooter":false,"format":"A4","newHeadless":true,"printBackground":true}}'" failed. Exit Code: 1(General error) Working directory: /home/erp/public_html/releases/20240123132720/public Output: ================ Error Output: ================ Error: Failed to launch the browser process! TROUBLESHOOTING: https://pptr.dev/troubleshooting at Interface.onClose (/home/erp/public_html/releases/20240123132720/node_modules/@puppeteer/browsers/lib/cjs/launch.js:267:24) at Interface.emit (node:events:529:35) at Interface.close (node:internal/readline/interface:534:10) at Socket.onend (node:internal/readline/interface:260:10) at Socket.emit (node:events:529:35) at endReadableNT (node:internal/streams/readable:1368:12) at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

The strange thing is that when I run this command on the Linux terminal with the same user as the server, it works. I get a stream of PDF back in the terminal:

PATH=$PATH:/usr/local/bin:/opt/homebrew/bin NODE_PATH=`npm root -g` node '/home/erp/public_html/releases/20240123132720/vendor/spatie/browsershot/src/../bin/browser.cjs' '{"url":"file:\/\/\/tmp\/1446216859-0747839001706019855\/index.html","action":"pdf","options":{"args":[],"viewport":{"width":800,"height":600},"displayHeaderFooter":false,"format":"A4","newHeadless":true,"printBackground":true}}'

Any ideas?

**Comment by ArielMejiaDev** [comment]
check your node installation with `wich node`, then try to add a browsershot closure to target your directories, if nothing helps, maybe is some kind of firewall in your browser, because puppeteer should work when the code is executed in the terminal.

**Comment by yannisc** [comment]
Unfortunately, nothing worked. I tried setting the node and npm paths to the 'which node' and 'which npm' paths, I also tried disabling the firewall, but nothing worked. At all times the command from the terminal seemed to work, but the browser responded with the same error message that I posted above.

**Comment by mrksdiehl** [comment]
I have **exactly** the same issue. I can not count how often I reinstalled and added multiple dependancies. It is also the same here: When I copy/paste the command from the error message and run it directly from the terminal, the command runs fine. The exact same command does not work when initiated by laravel.

**Comment by RemiHin** [comment]
I too have this issue. Can't for the life of me figure out how to fix it. I've tried almost everything it feels.

**Comment by mrksdiehl** [comment, marked as answer]
So, I managed to "fix" my issue. Maybe others have the same problem. I am running Laravel on Open Lite Speed.
First of all I tried to launch puppeteer's chrome directly from laravel to check if the issue is related to puppeteer or to chrome. That failed, so I knew it was no problem of the package.
Then I found this post https://www.litespeedtech.com/support/forum/threads/chrome-headless-crashes-with-cannot-allocate-memory-error.21545/
So I just did exactly what this guy did and increased the memory limit for lsphp to 80G (even though the server only has 16) and since then it is working just fine.

**Reply by yannisc** [reply]
Setting memory limit to 80G worked for me as well!
